src/common/exponent/exponentPlatform.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {ErrorHelper} from "../error/errorHelper";
+import {InternalErrorCode} from "../error/internalErrorCode";
+import {IRunOptions} from "../launchArgs";
+import {Log} from "../log/log";
+import {GeneralMobilePlatform} from "../generalMobilePlatform";
+
+import * as Q from "q";
+
+export class ExponentPlatform extends GeneralMobilePlatform {
+    private exponentTunnelPath: string;
+
+    constructor(runOptions: IRunOptions, { remoteExtension = null } = {}) {
+        super(runOptions, { remoteExtension: remoteExtension });
+        this.exponentTunnelPath = null;
+    }
+
+    public runApp(): Q.Promise<void> {
+        const outputMessage = `Application is running on Exponent. Open your exponent app at ${this.exponentTunnelPath} to see it.`;
+        Log.logMessage(outputMessage);
+        this.remoteExtension.showInformationMessage(outputMessage);
+        return Q.resolve<void>(void 0);
+    }
+
+    public enableJSDebuggingMode(): Q.Promise<void> {
+        Log.logMessage("Application is running on Exponent. Please shake device and select 'Debug JS Remotely' to enable debugging.");
+        return Q.resolve<void>(void 0);
+    }
+
+    public startPackager(): Q.Promise<void> {
```

Comment:
This is part of our debugging workflow, and could more accurately be called "ensurePackagerIsRunning". When debugging finishes we leave the packager running, and if the packager was running prior to debugging this call does not start a new one (Unless it is the wrong flavour of packager)